qcodes/instrument/parameter.py

```diff
@@ -1020,8 +1024,11 @@ def __init__(self, name: str,
             self.gettable = True
             self.get = self._wrap_get(self.get_raw)
 
-
-        if not hasattr(self, 'set') and set_cmd is not False:
+        if self.settable and set_cmd is not None:
```

Comment:
We probably need to allow set_cmd is False too to allow the default value to subclass